mmv1/products/vertexai/Models.yaml

```diff
@@ -0,0 +1,419 @@
+            properties:
+              - type: Array
+                name: 'command'
+                description: Command is the command line to execute inside the container, the working directory for the command is root ('/') in the container's filesystem.
+                item_type:
+                  type: String
+      - type: NestedObject
+        name: 'healthProbe'
+        immutable: true
+        description: Specification for Kubernetes readiness probe.
+        properties:
+          - type: Integer
+            name: 'periodSeconds'
+            immutable: true
+            default_value: 10
+            description: How often (in seconds) to perform the probe. Default to 10 seconds. Minimum value is 1. Must be less than timeoutSeconds.
+          - type: Integer
+            name: 'timeoutSeconds'
+            immutable: true
+            default_value: 1
+            description: Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1. Must be greater or equal to periodSeconds.
+          - type: NestedObject
+            name: 'exec'
+            description: Number of seconds after which the probe times out. Defaults to 1 second. Minimum value is 1. Must be greater or equal to periodSeconds.
+            properties:
+              - type: Array
+                name: 'command'
+                description: Command is the command line to execute inside the container, the working directory for the command is root ('/') in the container's filesystem.
+                item_type:
+                  type: String
+  - type: String
+    name: 'artifactUri'
+    immutable: true
+    default_from_api: true
+    description: |
+      The path to the directory containing the Model artifact and any of its supporting files.
+  - type: String
+    name: 'createTime'
+    output: true
+    description: |
+      The timestamp of when the Model was created in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits.
+  - type: String
+    name: 'updateTime'
+    output: true
+    description: |
+      The timestamp of when the MetadataStore was last updated in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits.
+  - type: String
+    name: 'modelName'
+    default_from_api: true
+    description:
+      The display name of the Model. The name can be up to 128 characters long
+      and can consist of any UTF-8 characters.
+  - type: String
+    name: 'etag'
+    description: Used to perform consistent read-modify-write updates.
+    output: true
+    ignore_read: true
+  - type: Array
+    name: 'versionAliases'
+    default_from_api: true
+    description:
+      user provided version aliases so that a model version can be referenced via alias.
+    item_type:
+      type: String
+  - type: KeyValueLabels
+    name: 'labels'
+    description: The labels with user-defined metadata to organize your Models.
```

Comment:
This makes since `labels` is meant to represent labels that are explicitly set in the terraform configuration.

We see the labels preset in effective labels and terraform labels since

- `effective_labels` represent all labels found on the resource
- `terraform_labels` represent labels that terraform is managing

So in the test configs we don't set labels (and aren't able to anyways). I'll remove it but keep in mind this will be in `:UPLOAD` since it's supported there.

Reply:
I accidentally posted about labels [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/12074#discussion_r1829883529), but should have posted here instead.


I realised that the labels field can be updated on a model created via copy, but they cannot be set at create time.

This means we're stuck between copy and update time:
- At create time the labels field would ideally be Computed, to pull down labels copied from the source model. The user cannot specify labels at this point.
- At update time, the user can specify labels to update the model. However they'd need to be careful to copy labels into their config that were originally stored in the API only.

I guess the solution here is the field is default_from_api:true, and we add an update after the create action (post_create) to set the labels. Or maybe we can add validation that stops users setting `labels` during create.

I think our options are to:
1) Remove labels from this PR, and let it be a FR added to the resource after it's initially released
2) Remove labels from this PR, and then add them in a subsequent PR on the feature branch. Maybe adding them after both copy and update are done?
3) Tackle the problem in this PR.

@BBBmau what's your preference here? I'm leaning towards 1 or 2

Reply:
Would lean towards 1, thanks for bringing this up!